lib/perl/Genome/Config/AnalysisProject/Command/DiskUsage.pm

```diff
@@ -0,0 +1,107 @@
+
+use strict;
+use warnings;
+
+use Genome;
+
+class Genome::Config::AnalysisProject::Command::DiskUsage {
+    is => 'Command::V2',
+    has => [
+        analysis_projects => {
+            is                  => 'Genome::Config::AnalysisProject',
+            doc                 => 'the analysis projects to report disk usage',
+            is_many             => 1,
+            shell_args_position => 1,
+        },
+    ],
+    doc => 'report on the disk usage of analysis projects',
+};
+
+sub help_detail {
+    return <<EOHELP
+This command will take several minutes to run.
+
+The output is a tab-delimited disk usage summary for each model type and configuration item per analysis project.
+The colums of the output file are:
+analysis project ID, configuration item ID, subclass name, model count, build count, total KB, total base pair, bytes per base
+
+EOHELP
+}
+
+sub execute {
+    my $self = shift;
+
+    my @data;
+    for my $anp ($self->analysis_projects) {
+        $self->status_message('Working on Analysis Project: '. $anp->__display_name__);
+        for my $config_item ($anp->config_items) {
+            $self->status_message('Working on Config Item: '. $config_item->id);
+            my %model_type_disk_usage;
+            my $model_count = 0;
+            my $build_count = 0;
+            for my $model ($config_item->models) {
+                $self->status_message('Working on Model: '. $model->__display_name__);
+                my $subclass_name = $model->subclass_name;
+                $model_type_disk_usage{$subclass_name}{'model_count'}++;
+                my $build = $model->last_succeeded_build;
+                if ($build) {
+                    $self->status_message('Working on Build: '. $build->id);
+                    $model_type_disk_usage{$subclass_name}{'build_count'}++;
+                    my @allocations = $build->disk_usage_allocations;
+                    for my $allocation (@allocations) {
+                        $model_type_disk_usage{$subclass_name}{'allocations'}{$allocation->id} = $allocation->kilobytes_used;
+                    }
+                    my @instrument_data = $build->instrument_data;
+                    for my $instrument_data (@instrument_data) {
+                        $model_type_disk_usage{$subclass_name}{'instrument_data'}{$instrument_data->id} = $instrument_data->total_bases_read;
+                    }
+                }
+            }
+            for my $model_type (sort keys %model_type_disk_usage) {
+                my $total_kb = 0;
+                map { $total_kb += $_ } values(%{$model_type_disk_usage{$model_type}{'allocations'}});
+
+                my $total_bp = 0;
+                map { $total_bp += $_ } values (%{$model_type_disk_usage{$model_type}{'instrument_data'}});
+
+                my $bytes_per_base = 'na';
+                if ($total_bp) {
+                    $bytes_per_base = ($total_kb * 1024) / $total_bp;
+                }
+                my %data = {
```

Comment:
This still isn't quite right.

`%data` expects `( ... )`.

This could also be `my $data = { ... } ; push @data, $data;`